R/allele_qc.R

```diff
@@ -41,48 +89,59 @@ allele_qc = function(sumstats,info_snp,match.min.prop,remove_dups,flip,remove) {
  	snp[["keep"]][ a2 != "A" & a2 != "T" & a2 != "G" & a2 != "C" ] = F
+    
  	# as long as scenario 1 is involved, sign_flip will return TRUE
  	snp[["sign_flip"]] = (a1 == ref2 & a2 == ref1) | (a1 == flip2 & a2 == flip1)
+    
  	# as long as scenario 2 is involved, strand_flip will return TRUE
  	snp[["strand_flip"]] = (a1 == flip1 & a2 == flip2) | (a1 == flip2 & a2 == flip1)
+    
  	# remove other cases, eg, tri-allelic, one dataset is A C, the other is A G, for example.
  	exact_match = (a1 == ref1 & a2 == ref2) 
  	snp[["keep"]][!(exact_match | snp[["sign_flip"]] | snp[["strand_flip"]])] = F
-   
-  matched = matched%>%
+
+    # Add the 'keep', 'sign_flip', and 'strand_flip' flags to the matched data
+    matched = matched%>%
             mutate(keep = snp[["keep"]])%>%
             mutate(sign_flip= snp[["sign_flip"]])%>%
             mutate(strand_flip=snp[["strand_flip"]])
-  if(flip) {
-    if(!is.null(matched$beta)) {
-    matched$beta[matched$sign_flip] = -1 * matched$beta[matched$sign_flip]
+    
+    # Apply allele flip if required
+    if(flip) {
+      if(!is.null(target_data_merged[,col_to_flip])){
+         target_data_merged[matched$sign_flip,col_to_flip] = -1*target_data_merged[matched$sign_flip,col_to_flip]
+      } else {
+         stop("Column '", col_to_flip, "' not found in target_data.")
+      }
     }
-    matched$beta[matched$sign_flip] = -1 * matched$beta[matched$sign_flip]
-    matched$z[matched$sign_flip] = -1 * matched$z[matched$sign_flip]
-    matched$A1.sumstats[matched$sign_flip] = matched$A1.ref[matched$sign_flip]
-    matched$A2.sumstats[matched$sign_flip] = matched$A2.ref[matched$sign_flip]
-  }  
-  if(remove) {
-    matched = matched[matched$keep,]
-  }
-  if (remove_dups) {
-    dups <- vec_duplicate_detect(matched[, c("chrom", "pos","A1.sumstats","A2.sumstats")])
-    if (any(dups)) {
-      matched <- matched[!dups, ]
-      message2("Some duplicates were removed.")
+    
+    # Keep SNPs based on the 'keep' flag
+    if(remove) {
+       target_data_merged = target_data_merged[matched$keep,]
     }
-  }
-  #match.min.prop Minimum proportion of variants in the smallest data to be matched, otherwise stops with an error. Default is `20%`.
-  min_match <- match.min.prop * min(nrow(sumstats), nrow(info_snp))
-  if (nrow(matched) < min_match)
-  stop("Not enough variants have been matched.")
-  return(matched)
+
+    # Remove duplicates if specified
+    if (remove_dups) {
+       dups <- vec_duplicate_detect(matched[, c("chrom", "pos","A1.target","A2.target")])
+       if (any(dups)) {
+          target_data_merged <- target_data_merged[!dups, ]
+          message2("Some duplicates were removed.")
+        }
+     }
+
+    # Check if the minimum proportion of variants is matched
+    min_match <- match.min.prop * min(nrow(target_variants), nrow(ref_variants))
+    if (nrow(target_data_merged) < min_match){
+       stop("Not enough variants have been matched.")
+     }
+     return(target_data_merged)
```

Comment:
I think you can return two things in a list: 1. target_data_qced, 2. qc_summary, where 1 is your current output, 2 is the qc summary table?